tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -1097,6 +1097,70 @@ def forward_xp(self, inputs, xp):
         return xp.tanh(x),
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@pytest.mark.parametrize('input', [
+    numpy.asarray(0), numpy.asarray(-1), numpy.asarray(1), numpy.asarray(
+        10), numpy.full((), 2), numpy.full((0,), 2), numpy.full((2, 3), 2),
+    numpy.random.randn(2, 4), numpy.random.randn(4, 4),
+    numpy.random.randn(1, 4)
+])
+@pytest.mark.parametrize('contiguous', [None, 'C'])
+@pytest.mark.parametrize('in_dtype,out_dtype', _expected_dtypes_math_functions)
+class TestSin(op_utils.NumpyOpTest):
```

Comment:
How about using `UnaryMathTestBase` as the base class? It seems to reduce code duplication.

Reply:
Makes sense. Will do that.